### PR TITLE
Added After Transition on Shipment

### PR DESCRIPTION
**Pull request by patvice**
@kurtfunai @uri @glongman please review

### CODE REVIEW

**Comment by glongman** [comment]
lgtm. I think a new test in the main app would be good too.

File: core/app/models/spree/shipment.rb [hunk 1]
```diff
@@ -37,6 +37,10 @@ class Shipment < ActiveRecord::Base
           shipment.determine_state(shipment.order) == 'ready'
         }
       end
+      after_transition from: :pending, to: :ready, do: lambda{ |shipment|
+        # .pickup? is defined in the HW ShippingMethod Decorator
+        shipment.ship! if shipment.shipping_method.try(:pickup?)
+      }
```
Comment (kurtfunai) [inline core/app/models/spree/shipment.rb:43]: I think this looks good as long as the decorator from HW is loaded correctly :)

Comment (patvice) [inline core/app/models/spree/shipment.rb:43]: It did when I was testing locally

**Comment by patvice** [comment]
good idea @glongman, i'll write some in the local pickup branch

**Comment by glongman** [comment]
uri may have found something on staging that you need to know before merge

**Comment by uri** [comment]
If I'm correct this won't fix the problem. That being said, just shoot it up onto staging and give it a quick try.

**Comment by patvice** [comment]
Closed in favour of just leaving the functionality in the `Spree::Order#finalize!`

**Pull request closed without merging**